docs/adr/adr-007-data-retention.md

```diff
@@ -0,0 +1,196 @@
+# Decision Record: Data Retention
```

Comment:
Looks like this is missing a number at the start, eg `7. Data Retention` as per [the template](https://github.com/alphagov/email-alert-api/blob/master/docs/adr/adr-005-record-architecture-decisions.md) and it's not necessary to state "Decision Record:".

I also notice [my ADR](https://github.com/alphagov/email-alert-api/pull/1476) is also flagged as number 7 🤦  I'll update mine to 8. Gosh it's been a while since I've been a project with concurrent ADR's - it's like being back on Publishing Workflow 😁 .

Reply:
Haha ADR's everywhere 👀 😆